### PR TITLE
Tracking: add Cell Tracking Challenge export format

### PR DESCRIPTION
**Pull request by chaubold**
This PR adds functionality to export the tracking result in the format defined by the [ISBI Cell Tracking Challenge](http://www.codesolorzano.com/Challenges/CTC/Welcome.html). Check [here](http://ctc2015.gryf.fi.muni.cz/Public/Documents/Naming%20and%20file%20content%20conventions.pdf) for a detailed description of the file contents.

It also fixes the export filename for tracking plugins that take the name as folder instead of file.

**Update 25.10.2017:**
* fixed a problem with mergers in JSON export
* added a bunch of tests that at least runs our plugins and also verifies the result for some of them

### CODE REVIEW

**Comment by chaubold** [comment]
(Why are our AppVeyor tests timing out so often? Can we maybe somehow cache the conda environment, as we do on CircleCI as well?)

**Comment by k-dominik** [comment]
> (Why are our AppVeyor tests timing out so often? Can we maybe somehow cache the conda environment, as we do on CircleCI as well?)

you have deferred correctly, that we are not caching the conda env right now. Didn't even suspect that the builds might take that long. I'll look into it

File: ilastik/plugins_default/tracking_ctc_export.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    def _save_tracks(self, tracks, output_dir):
+        """
+        Take a dictionary indexed by TrackId which contains
+        a list [parent, begin, end] per track, and save it 
+        in the text format of the cell tracking challenge.
+        """
+        filename = os.path.join(output_dir, 'res_track.txt')
+        with open(filename, 'wt') as f:
+            for key, value in tracks.items():
+                if key ==  None:
```
Comment (k-dominik) [inline ilastik/plugins_default/tracking_ctc_export.py]: `is None`... pep8?

File: ilastik/plugins_default/tracking_ctc_export.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+        :param filename: string of the FILE where to save the result (will be appended with _graph.json and _result.json)
+        :param hypothesesGraph: hytra.core.hypothesesgraph.HypothesesGraph filled with a solution
+        :param objectFeaturesSlot: lazyflow.graph.InputSlot, connected to the RegionFeaturesAll output
+               of ilastik.applets.trackingFeatureExtraction.opTrackingFeatureExtraction.OpTrackingFeatureExtraction
+
+        :returns: True on success, False otherwise
+        """
+        if not os.path.exists(filename):
+            os.makedirs(filename)
```
[the remark this answers is not in the document]
Comment (chaubold) [reply ilastik/plugins_default/tracking_ctc_export.py]: Oh yeah, indeed. Thx

File: ilastik/plugins_default/tracking_ctc_export.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                self._save_frame_to_tif(timeframe, labelFrame, filename)
+
+        return True
+
+
+    def _save_frame_to_tif(self, timestep, label_image, output_dir, filename_zero_padding=3):
+        """
+        Save a single frame to a 2D or 3D tif
+        """
+        filename = os.path.join(output_dir, 'mask' + format(timestep, "0{}".format(filename_zero_padding)) + '.tif')
```
[the remark this answers is not in the document]
Comment (chaubold) [reply ilastik/plugins_default/tracking_ctc_export.py]: Good suggestion! That line dates back to the Python 2 era...

**Comment by k-dominik** [comment]
Cool, I assume, you have tried it at least once after the latest additions...

LGTM

**Comment by chaubold** [comment]
Yes, but I guess a test couldn't hurt. I'll set up a test that at least runs each plugin...

**Comment by k-dominik** [comment]
Okay, I do trust the tests, LGTM

**Pull request merged**